src/foam/mlang/mlang.js

```diff
@@ -2380,7 +2382,9 @@ while ( i.hasNext() ) {
 
   try {
     String s = "";
-    if ( prop instanceof foam.core.AbstractFObjectPropertyInfo ) {
+    if ( ReferenceExpressionHelper.isPropertyAReference((PropertyInfo)prop) ) {
```

Comment:
I've added backend support but I'm not sure what to do with front-end as js side changes must synchronous and I'm not sure how to implement that.
@kgrgreer @jlhughes please let me know what you think